tests/providers/amazon/aws/hooks/test_s3.py

```diff
@@ -313,6 +312,22 @@ def test_copy_object_acl(self, s3_bucket):
             assert ((response['Grants'][0]['Permission'] == 'FULL_CONTROL') and
                     (len(response['Grants']) == 1))
 
+    @mock_s3
+    def test_delete_bucket_if_bucket_exist(self, s3_bucket):
+        # assert if the bucket is created
+        mock_hook = S3Hook()
+        mock_hook.create_bucket(bucket_name=s3_bucket)
+        assert mock_hook.check_for_bucket(bucket_name=s3_bucket)
+        mock_hook.delete_bucket(bucket_name=s3_bucket, force_delete=True)
+        assert not mock_hook.check_for_bucket(s3_bucket)
+
+    @mock_s3
+    def test_delete_bucket_if_not_bucket_exist(self, s3_bucket):
+        # assert if exception is raised if bucket not present
+        mock_hook = S3Hook()
+        with pytest.raises(Exception):
```

Comment:
This is too broad an exception to catch! https://boto3.amazonaws.com/v1/documentation/api/latest/guide/error-handling.html?highlight=exception#catching-exceptions-when-using-a-resource-client

Reply:
NoSuchBucket error is raised when there is a ClientError, there is no specific exception for NoSuchBucket.

Reply:
Oh missed that one. 

> NoSuchBucket error is raised when there is a ClientError

Sounds good. Then you can check for `ClientError`  and check the message or some kind of code.
Yes, like in the second example on the link ash provided.